### ISSUE TITLE
Consider adding a default handler to map spray rejections to JSend

### ISSUE THREAD

**Issue by gabro** [buildo/nozzle#8]
Currently in our internal projects we provide custom rejection handlers that produce a `JSendFailure`, then we leave the rest to the default rejection handler, which produces meaningful responses, with appropriate HTTP error codes, but that is in plaintext.

It should be trivial to add a default JSend wrapper for the default rejection handler response, using `mapHttpResponse`. Something along the lines of

``` scala
  def toJSend(res: HttpResponse): HttpResponse = 
    res.withEntity(HttpBody(ContentType.`application/json`,
      JSendFailure(message = res.entity.asString.toJson).toJson))

  implicit val jsonRejectionHandler = RejectionHandler {
    case rejections => mapHttpResponse(toJSend)(RejectionHandler.Default(rejections))
  }
```

Not sure the above compiles, but you get the idea.

@utaal, thoughts?

I know @federico-pellegatta is trying to implement this in one of our projects.

**Comment by utaal** [comment]
Sure this isn't in the new Scala base already (for monadic controllers)? I
have to check.

**Comment by utaal** [comment]
Possibly incomplete, but have a look at https://github.com/buildo/labonline/blob/master/escalapio/src/main/scala/labonline/routers/EscalapioRouterModule.scala#L79

**Comment by gabro** [comment]
@utaal not quite. Here's what I mean: https://github.com/buildo/labonline/commit/edd47c0873683eac6ec5da9c19f1036620de7b74

That leverages the default messages and codes already defined in spray-routing, simply wrapping the in a `JSendFailure`

**Comment by gabro** [comment]
I think we have something like this in master now, right? @utaal

**Comment by utaal** [comment]
Maybe. But if it's not there we should make it happen.
